frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -186,25 +170,22 @@ export function InstantSearchProvider({
             const deviceHandle = pathParts.length >= 2 ? pathParts[1] : '';
             const itemType = pathParts.length >= 3 ? pathParts[2] : '';
 
-            const {
-               q = '',
-               p,
-               filter = {},
-               // range = {},
-            } = qsModule.parse(location.search.slice(1));
+            const { q, p, filter } = qsModule.parse(location.search.slice(1));
+
+            const filterObject = (
+               typeof filter === 'object' ? filter : {}
```

Comment:
I think this was included to cover the case where `filter` is a string, since the return type of `qaModule.parse` is `ParsedQs`.

```ts
interface ParsedQs { [key: string]: undefined | string | string[] | ParsedQs | ParsedQs[] }
```

I think this would work as well, without the need for the type assertion.

```ts
            const filterObject =
               typeof filter === 'object' && !Array.isArray(filter)
                  ? filter
                  : {};
```

Reply:
Maybe we could use `isPlainObject` by lodash? Since the library is already in the project and we can use it without adding too much overweight
https://lodash.com/docs/4.17.15#isPlainObject

Reply:
For simple expressions like this, I think I prefer using the idiomatic structure. If I was reading this code, I'd be able to understand `typeof X == 'object'` much faster than `isPlainObject()` I'd probably *guess* about what `isPlainObject()` does and I'd likely guess right, but seeing it would make me think "It was complex enough to extract into a function and thus probably more than just `typeof`".

But yes, the library function also protects from typos in `object`, so there's that. Hmm.. Though I think typescript protects us from typos too.

I'm not adamant either way, but I don't don't think one is so much better than the other that it warrants changing.

Oof, when reading the description and the source I'm even less sure if it's what I want.
![image](https://user-images.githubusercontent.com/26855/191567540-bbc9afbf-99b4-4d72-b46f-43c7e92409ee.png)

That said, checking for an array instead of just pretending it's not possible, so I'll work on that.